Sources/StreamChatUI/ChatChannelList/SwipeableView.swift

```diff
@@ -153,11 +153,21 @@ open class SwipeableView: _View, ComponentsProvider, UIGestureRecognizerDelegate
         //
         // *This practically means on panning the cell we don't accidentally scroll the list
         let translation = recognizer.translation(in: self)
+        if abs(translation.x) <= abs(translation.y) {
+            return false
+        }
 
-        return abs(translation.x) > abs(translation.y)
+        // Additionally, if the horizontal swipe is left-to-right and action items
+        // are not expanded, we'll deny this recognizer, so that it doesn't steal
+        // the gesture from another recognizer, such as a UINavigationController's
+        // interactivePopGestureRecognizer
+        if !isOpen, translation.x > 0 {
+            return false
+        }
```

Comment:
In order to support swipe-left-to-right actions, we can actually define a threshold for the origin of the swipe 🤔 If the swipe is too close to the edge, we return false, but if it is not, we don't ignore it. (This is how I believe for example how the Mail app from Apple works)

The thing is that, we (The SDK) by default do not support swipe-left-to-right actions, but another customer might i've added swipe-left-to-right actions, and this would prevent it to work. 

If you do not have the time to add this improvement, I can take it from here ofc 🙂 This already very helpful, Thank you! 🙏

Reply:
Thanks @nuno-vieira! I wasn't considering the fact that a client may have added left-to-right actions.
I'll let you take it from here. We've implemented a work-around, so prioritize this as you wish.